### PR TITLE
Fix crash when passing null to clearImmediate

### PR DESCRIPTION
**Pull request by janicduplessis**
Passing `undefined` or `null` to `clearImmediate` caused apps to crash. It it caused because we try to find the index of the null/undefined timer when we should just do nothing when passed these values.

It is already handled properly in the other Timer functions.

**Test plan**
Calling `clearImmediate` with `undefined` or `null` should do nothing.

### CODE REVIEW

**Comment by vjeux** [comment]
@facebook-github-bot shipit

Good catch!

**Comment by janicduplessis** [comment]
@vjeux I think the bot is sleeping :(

File: Libraries/JavaScriptAppEngine/System/JSTimers/JSTimers.js [hunk 1]
```diff
@@ -106,10 +106,12 @@ var JSTimers = {
 
   clearImmediate: function(timerID) {
     JSTimers._clearTimerID(timerID);
-    JSTimersExecution.immediates.splice(
-      JSTimersExecution.immediates.indexOf(timerID),
-      1
-    );
+    if (timerID != null) {
```
Comment (nicklockwood) [inline Libraries/JavaScriptAppEngine/System/JSTimers/JSTimers.js]: Shouldn't this be `!==` ? @vjeux

Comment (janicduplessis) [inline Libraries/JavaScriptAppEngine/System/JSTimers/JSTimers.js]: We want `!=` here so it works will `null` and `undefined`. It's the same check the is used in other places in this file.

**Comment by vjeux** [comment]
Actually, it would still crash if you clearImmediate a value that has already been removed.

Could you do

```
var index = x.indexOf()
if (index !== -1) {
  x.splice()
```

**Comment by vjeux** [comment]
why do you still check for timer != null? Index === -1 check should catch this no?

**Comment by janicduplessis** [comment]
I wasn't sure if there could be null values in the immediates array since there was a check for it for the timerID array (https://github.com/janicduplessis/react-native/blob/clear-immediate/Libraries/JavaScriptAppEngine/System/JSTimers/JSTimers.js#L120). I double checked if it was possible and it is not so you are right I can remove the null check.

**Comment by vjeux** [comment]
@facebook-github-bot shipit

**Pull request closed without merging**